server/routerlicious/packages/services-utils/src/tokenManager.ts

```diff
@@ -0,0 +1,109 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import {
+	IWebSocket,
+	IWebSocketTracker,
+	IJsonWebTokenManager,
+} from "@fluidframework/server-services-core";
+import { Lumberjack } from "@fluidframework/server-services-telemetry";
+import { NetworkError } from "@fluidframework/server-services-client";
+
+export class WebSocketTracker implements IWebSocketTracker {
+	// Map of socket id to socket object
+	private readonly socketIdToSocketMap: Map<string, IWebSocket>;
+	// Map of composite token id to socket ids. It assumes one token could be used for multiple sockets
+	private readonly tokenIdToSocketIdMap: Map<string, Set<string>>;
+	// Map of socketId to token ids. It assumes one socket could be used for connections with multiple tokens
+	private readonly socketIdToTokenIdMap: Map<string, Set<string>>;
+
+	constructor() {
+		this.socketIdToSocketMap = new Map();
+		this.tokenIdToSocketIdMap = new Map();
+		this.socketIdToTokenIdMap = new Map();
+	}
+
+	public addSocket(compositeTokenId: string, webSocket: IWebSocket) {
+		if (this.tokenIdToSocketIdMap.has(compositeTokenId)) {
+			this.tokenIdToSocketIdMap.get(compositeTokenId)?.add(webSocket.id);
+		} else {
+			this.tokenIdToSocketIdMap.set(compositeTokenId, new Set([webSocket.id]));
+		}
+
+		if (this.socketIdToTokenIdMap.has(webSocket.id)) {
+			this.socketIdToTokenIdMap.get(webSocket.id)?.add(compositeTokenId);
+		} else {
+			this.socketIdToTokenIdMap.set(webSocket.id, new Set([compositeTokenId]));
+		}
+
+		this.socketIdToSocketMap.set(webSocket.id, webSocket);
+	}
+
+	public getSockets(compositeTokenId: string): IWebSocket[] {
+		const socketIds = this.tokenIdToSocketIdMap.get(compositeTokenId);
+
+		if (!socketIds) {
+			return [];
+		}
+
+		const socketResult: IWebSocket[] = [];
+		for (const socketId of socketIds) {
+			const socketObj = this.socketIdToSocketMap.get(socketId);
+			if (socketObj) {
+				socketResult.push(socketObj);
+			}
+		}
```

Comment:
Use `map()` function to simplify the code?

Reply:
socketIds is a Set and Set doesn't have map function. I make it Set instead of array because I want to make sure it doesn't have duplicates. I can use forEach but it doesn't simplify the code too much.